Increase transparency of red ice in CTMC

Hexocyte opened the pull request:
When "Chest Appearance Matches Contents" is enabled, the transparency of red ice, except red ice walls, is increased to make it easier to see the chest textures.

Before:
![screenshot](https://user-images.githubusercontent.com/9218741/174408018-2143aef6-477e-4f39-8ead-40c29c75ebdc.png)

After:
![screenshot](https://user-images.githubusercontent.com/9218741/174408063-e54b8cc3-d005-4dd5-8a3e-4ecb89512a0e.png)

Tested in Dolphin 5.0-16101 and RetroArch/Mupen64Plus-Next.

Hexocyte commented:
I'm figuring out how to add ASM hacks, so I'm redoing this so that it only affects the red ice crystals with chests.

Hexocyte commented:
I made it specific to the red ice crystals with chests and lowered the opacity slightly further. I updated the PR title and description above.

cjohnson57 commented:
Hi! Unfortunately this PR is a bit out of date now, if you update it to latest I'll probably approve it.

r0bd0g commented:
Now that we're changing the chest textures again, is this change still necessary?

Hexocyte commented:
If I updated this and still put all the CTMC code in Patches.py into an if block as I originally did, it'd conflict with #1559. I think the big if block makes it neater, so I'll wait until that PR is merged before I change this.

In #1502 chests are drawn in front of red ice and flame circles (even when Invisible Chests is off). That might be a better way to improve visibility? Maybe not, since it can look like the chest really is in front of the ice.

mracsys commented on ASM/src/hacks.asm:
```diff
@@ -1238,6 +1238,14 @@ skip_GS_BGS_text:
+
+; Replaces addiu    t7, $zero, 0x00FF
+.orga 0xDB3244
+    j       red_ice_alpha
```
What is the next instruction in the delay slot? Is that affected by any of the new code?

This should also be jal, not j.

Hexocyte replied:
The instruction in the delay slot sets the action function and is unaffected. I added a comment with the instruction.

I used j instead of jal as a shortcut to avoid having to push the old ra to the stack. I don't think there's enough room in the function to do that properly. red_ice_alpha does the skipped instructions instead.

mracsys commented on Patches.py:
```diff
@@ -1743,40 +1743,44 @@ def update_scrub_text(message, text_replacement, default_price, price, item_name
-            if item['chest_type'] in (BROWN_CHEST, SILVER_CHEST, SKULL_CHEST_SMALL):
-                rom.write_int16(chest_address_0 + 6, 0x0172)  # Z pos
-                rom.write_int16(chest_address_2 + 6, 0x0172)  # Z pos
+    if world.settings.correct_chest_appearances != 'off':
```
No need for putting the CSMC/CTMC changes in the same if block as the red ice changes in my opinion. Also makes it easier to merge with other open PRs if separated.

Hexocyte replied:
I removed the Python changes when changing it to also affect red ice without chests.

mracsys commented on Patches.py:
```diff
@@ -2172,7 +2176,19 @@ def set_spirit_shortcut(rom, actor_id, actor, scene):
+    ice_z_coords = (
+        0xF5D8, # z=-2600 Ice Cavern Map Chest
+        0x0283, # z=283   Ice Cavern Compass Chest
+        0xFB80, # z=-1152 Ganons Castle MQ Water Trial Chest
```
In my opinion, transparency changes should be generic for all red ice blocks, not just for frozen chests.

cjohnson57 replied:
I agree with this.

Hexocyte replied:
Done. It now affects all red ice except red ice walls.

mracsys reviewed (commented):
Has this been tested on console?

Hexocyte commented:
Now I made it so that it applies to all red ice except red ice walls. I also added more comments and tweaked the opacity again.

Hexocyte commented:
> Now that we're changing the chest textures again, is this change still necessary?

I think it still helps. The front of the major item chest didn't change much, and I figure even with improved textures, it's bound to be at least a little hard sometimes to distinguish the chest types in full-opacity red ice.

Hexocyte committed:
Increase transparency of red ice in CTMC

Increase red ice transparency when "Chest Appearance Matches Contents"
is enabled to make it easier to see the chest textures. Red ice walls
are excepted and remain opaque.
